yarn-project/noir-protocol-circuits/src/crates/types/src/abis/block_header.nr

```diff
@@ -1,40 +1,84 @@
-use crate::block::Block;
+use crate::{
+    constants::{
+        BLOCK_HEADER_LENGTH,
+        GENERATOR_INDEX__BLOCK_HASH,
+    },
+    hash::pedersen_hash,
+};
 
+// docs:start:block-header
 struct BlockHeader {
-    archive_root : Field,
-    block : Block,
-    // Private data
-    // This is marked in the cpp code as an enhancement
-    private_kernel_vk_tree_root : Field,
+    note_hash_tree_root : Field,
+    nullifier_tree_root : Field,
+    contract_tree_root : Field,
+    l1_to_l2_messages_tree_root : Field,
+    archive_root: Field,
+    public_data_tree_root: Field,
+    global_variables_hash: Field,
 }
+// docs:end:block-header
 
 impl BlockHeader {
-    fn assert_is_zero(self) {
-        self.block.assert_is_zero();
-        assert(self.private_kernel_vk_tree_root == 0);
+    pub fn assert_is_zero(self) {
+        assert(self.note_hash_tree_root == 0);
+        assert(self.nullifier_tree_root == 0);
+        assert(self.contract_tree_root == 0);
+        assert(self.l1_to_l2_messages_tree_root == 0);
+        assert(self.archive_root == 0);
+        assert(self.public_data_tree_root == 0);
     }
 
-    fn to_array(self) -> [Field;7] {
+    pub fn serialize(self) -> [Field; BLOCK_HEADER_LENGTH] {
         // This comment was copied from the cpp codebase.
         //
         // TODO(#3441): Note private_kernel_vk_tree_root, is not included yet as
         // it is not present in noir,
         [
-            self.block.note_hash_tree_root,
-            self.block.nullifier_tree_root,
-            self.block.contract_tree_root,
-            self.block.l1_to_l2_messages_tree_root,
+            self.note_hash_tree_root,
+            self.nullifier_tree_root,
+            self.contract_tree_root,
+            self.l1_to_l2_messages_tree_root,
             self.archive_root,
-            self.block.public_data_tree_root,
-            self.block.global_variables_hash
+            self.public_data_tree_root,
+            self.global_variables_hash
         ]
     }
+
+    pub fn deserialize(deserialized: [Field; BLOCK_HEADER_LENGTH]) -> Self {
+        BlockHeader {
+            note_hash_tree_root: deserialized[0],
+            nullifier_tree_root: deserialized[1],
+            contract_tree_root: deserialized[2],
+            l1_to_l2_messages_tree_root: deserialized[3],
+            archive_root: deserialized[4],
+            public_data_tree_root: deserialized[5],
+            global_variables_hash: deserialized[6],
+        }
+    }
     
-    fn note_hash_tree_root(self) -> Field {
-        self.block.note_hash_tree_root
+    pub fn note_hash_tree_root(self) -> Field {
+        self.note_hash_tree_root
+    }
+
+    pub fn contract_tree_root(self) -> Field {
+        self.contract_tree_root
+    }
+
+    pub fn block_hash(self) -> Field {
```

Comment:
This is not really the block hash, but will be changed when we are doing the refactor in  https://github.com/AztecProtocol/aztec-packages/issues/3533